src/experimental/shader-modules/index.js

```diff
@@ -1,16 +1,15 @@
-import {registerShaderModules} from '../shader-tools';
+// import * as fp64 from './fp64';
 
-import * as fp64 from './fp64';
-
-import * as fog from './fog';
-import * as picking from './picking';
+// import * as fog from './fog';
+// import * as picking from './picking';
 // import * as lighting from './lighting';
-import * as material from './material';
+// import * as material from './material';
 
-registerShaderModules([
-  fp64,
-  fog,
-  picking,
-  // lighting,
-  material
-]);
+// TODO - This probably defeats tree-shaking
```

Comment:
> Or is there something I'm missing?

Well, index.js imports all files. Any statements on the top or "global" level will be executed during that import.

Tree shaking only comes into play after everything has been imported (in the case of webpack it doesn't happen until minification). The bundler starts looking for things that are not being imported/used. But then it is already too late. The bundler can no longer tree shake these things since they have indeed been used by the program.

In fact, this type of side effects is an huge unsolved problem for the tree-shakers and defeats much of their value for code that hasn't been carefully tuned.

As an example, even just setting a static member on a class (or the mere transpilation of the class which generates assignments to the prototype) can generate side effects that disable tree-shaking. 

That's why I added the babel plugin that injects /* PURE */ comments but also that prevented me from removing comments from the code, so I will need to find a configurable comment remover, sigh...